src/transformers/training_args_tf.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from dataclasses import dataclass, field
+from typing import Tuple
+
+from .file_utils import cached_property, is_tf_available, tf_required
+from .training_args import TrainingArguments
+
+
+logger = logging.getLogger(__name__)
+
+if is_tf_available():
+    import tensorflow as tf
+
+
+@dataclass
+class TFTrainingArguments(TrainingArguments):
+    optimizer_name: str = field(
+        default="adam",
+        metadata={
+            "help": 'Name of a Tensorflow optimizer among "adadelta, adagrad, adam, adamax, ftrl, nadam, rmsprop, sgd, adamw"'
+        },
+    )
+    mode: str = field(
+        default="sequence-classification",
+        metadata={"help": 'Type of task, one of "sequence-classification", "token-classification" '},
+    )
+    loss_name: str = field(
+        default="SparseCategoricalCrossentropy",
+        metadata={
+            "help": "Name of a Tensorflow loss. For the list see: https://www.tensorflow.org/api_docs/python/tf/keras/losses"
+        },
+    )
+    eval_steps: int = field(default=1000, metadata={"help": "Run an evaluation every X steps."})
+    debug: bool = field(
+        default=False, metadata={"help": "Activate the trace to record computation graphs and profiling information"}
+    )
+
+    @cached_property
+    @tf_required
+    def _setup_strategy(self) -> Tuple["tf.distribute.Strategy", int]:
+        logger.info("Tensorflow: setting up strategy")
+        gpus = tf.config.list_physical_devices("GPU")
+
+        if self.no_cuda:
+            strategy = tf.distribute.OneDeviceStrategy(device="/cpu:0")
+        else:
+            try:
+                tpu = tf.distribute.cluster_resolver.TPUClusterResolver()
+            except ValueError:
+                tpu = None
+
+            if tpu:
+                tf.config.experimental_connect_to_cluster(tpu)
+                tf.tpu.experimental.initialize_tpu_system(tpu)
+
+                strategy = tf.distribute.experimental.TPUStrategy(tpu)
+            elif len(gpus) == 0:
+                strategy = tf.distribute.OneDeviceStrategy(device="/cpu:0")
+            elif len(gpus) > 1:
+                # If you only want to use a specific subset of GPUs use `CUDA_VISIBLE_DEVICES=0`
+                strategy = tf.distribute.MirroredStrategy(gpus)
+            else:
+                raise ValueError("Cannot find the proper strategy please check your environment properties.")
+
+        return strategy
+
+    @property
+    @tf_required
+    def strategy(self) -> "tf.distribute.Strategy":
+        return self._setup_strategy
+
+    @property
+    @tf_required
+    def n_gpu(self) -> int:
+        return self._setup_strategy.num_replicas_in_sync
```

Comment:
Will this return the number of TPU cores when on a device having access to a TPU?

Reply:
Yes

Reply:
I'm thinking it should return `0` as there's no GPU. Is this property used in places where the number of TPU cores would change behavior?

Reply:
Here it returns the number of GPUs if the GPUs are used or TPUs if they are used

Reply:
Right, I guess this can be okay for the moment but down the road it would be nice to see have something clearer as the `n_gpu` property doesn't make much sense in a TPU environment.

I guess similar to the logging where it outputs `n_gpu: %s, distributed training: %s, 16-bits training: %s`, I guess we'd want to see `TPU cores: xxx ...`.

This can be in a next PR, it's mostly nit-picking!

Reply:
Ok, we can review that later indeed as I think we need a bit of discuss deeper how harmonize

Reply:
I can rename it to num_devices everywhere if it makes sense?

Reply:
Or accelerators? As you wish but indeed will have more sense to generalize here.